src/Umbraco.Web.UI.Client/src/packages/user/current-user/conditions/group-id/group-id.condition.ts

```diff
@@ -0,0 +1,65 @@
+import { UMB_CURRENT_USER_CONTEXT } from '../../current-user.context.token.js';
+import type { UmbCurrentUserModel } from '../../types.js';
+import type { UmbCurrentUserGroupIdConditionConfig } from './types.js';
+import { stringOrStringArrayIntersects } from '@umbraco-cms/backoffice/utils';
+import type { UmbControllerHost } from '@umbraco-cms/backoffice/controller-api';
+import type { UmbConditionControllerArguments, UmbExtensionCondition } from '@umbraco-cms/backoffice/extension-api';
+import { UmbConditionBase } from '@umbraco-cms/backoffice/extension-registry';
+
+export class UmbCurrentUserGroupCondition
+	extends UmbConditionBase<UmbCurrentUserGroupIdConditionConfig>
+	implements UmbExtensionCondition
+{
+	constructor(host: UmbControllerHost, args: UmbConditionControllerArguments<UmbCurrentUserGroupIdConditionConfig>) {
+		super(host, args);
+
+		this.consumeContext(UMB_CURRENT_USER_CONTEXT, (context) => {
+			this.observe(context.currentUser, this.observeCurrentUser, 'umbCurrentUserGroupConditionObserver');
+		});
+	}
+
+	private observeCurrentUser = async (currentUser: UmbCurrentUserModel) => {
+		const { match, oneOf, allOf, noneOf } = this.config;
+
+		if (match) {
+			if (currentUser.userGroupIds.includes(match)) {
+				this.permitted = true;
+				return;
+			}
+
+			this.permitted = false;
+			return;
+		}
+
+		if (oneOf) {
+			if (stringOrStringArrayIntersects(oneOf, currentUser.userGroupIds)) {
```

Comment:
I think for clarity(avoiding future complexity) we should just handle this as an array for now. Thought the utility is handy, there might be other places where we already do something alike where it can be used.

Reply:
Cause then (and that can be later) these lines of 24–61 could be turned into a utility method, cause this is a very general implementation that we could use across conditions :-)

Reply:
Also that one day where we make this a generic shared method, we could consider combining these checks... so you could use: match + oneOf ?

Reply:
It is currently an array:

```ts
	/**
	 * The user group(s) that the current user must be a member of to pass the condition.
	 * @examples [['guid1', 'guid2']]
	 */
	oneOf?: Array<string>;
```